cmd/kops-controller/main.go

```diff
@@ -324,3 +315,29 @@ func addGossipController(mgr manager.Manager, opt *config.Options) error {
 
 	return nil
 }
+
+func setupCloudIPAM(mgr manager.Manager, opt *config.Options) error {
+	setupLog.Info("enabling IPAM controller")
+	switch opt.Cloud {
+	case "aws":
+		ipamController, err := controllers.NewAWSIPAMReconciler(mgr)
+		if err != nil {
+			return fmt.Errorf("error creating aws IPAM controller: %w", err)
+		}
+		if err := ipamController.SetupWithManager(mgr); err != nil {
```

Comment:
Bottom-factor this out of the switch statement?

Reply:
Had to define an interface, but - done!